src/eval.rs

```diff
@@ -197,10 +285,147 @@ impl Expr {
             ExprSource::Implication { left, right } => vec![left, right],
             ExprSource::UnaryInvert { value } => vec![value],
             ExprSource::UnaryNegate { value } => vec![value],
+            ExprSource::AttrSet {
+                definitions,
+            } => {
+                let mut out = vec![];
+                out.extend(definitions);
+                // This looks similar to code at the end of the function, but
+                // we have Gc instead of Box, so we can't just return a vec
+                // like the rest of the `match` arms.
+                return out
+                    .into_iter()
+                    .map(|x| x.as_ref())
+                    .filter_map(Result::ok)
+                    .map(|x| x.as_ref())
+                    .collect();
+            }
+            ExprSource::KeyValuePair { key, value } => {
+                let mut out = vec![];
+                if let Ok(x) = value {
+                    out.push(x.as_ref());
+                }
+                if let Ok(x) = key {
+                    if let ExprSource::Dynamic { inner: Ok(val) } = &x.source {
+                        out.push(val.as_ref());
```

Comment:
This is only to provide information about values to the user, correct? Because, it seems a bit weird to push the key after the value if it's dynamic :)

Reply:
Yep, it's only for user information. I don't think the ordering of key vs value pushing matters here, since their ranges don't overlap (hovering over the key on non-dynamic attrs shows the value because the evaluator interprets the user as hovering over the entire attr set, which evaluates to the value)

Reply:
OK, fine! :)